pkg/utils/goroutine_pool.go

```diff
@@ -0,0 +1,33 @@
+package utils
+
+import "sync"
+
+type GoroutinePool struct {
+	pool      chan int
+	waitGroup *sync.WaitGroup
+}
+
+func NewGoroutinePool(size int) *GoroutinePool {
```

Comment:
The pool seems lack of func to be passed to work in goroutine

Reply:
I fixed it to store func and increase versatility.
To control the number of routines, I must use the “pool” prompt for completing the routine